src/Cake.Common/Tools/NuGet/Pack/NuGetPacker.cs

```diff
@@ -175,6 +176,18 @@ private ProcessArgumentBuilder GetArguments(FilePath filePath, NuGetPackSettings
                 builder.Append(settings.Verbosity.Value.ToString().ToLowerInvariant());
             }
 
+            // Properties
+            if (settings.Properties != null && settings.Properties.Count > 0)
+            {
+                if (settings.Properties.Values.Any(x => x == null))
+                {
+                    throw new CakeException("Properties values can not be null.");
+                }
+                builder.Append("-Properties");
+                builder.Append(string.Join(";",
+                    settings.Properties.Select(property => property.Key + "=" + property.Value)));
```

Comment:
My preference actually would have been to use string interpolation, but I didn't see any other C# 6 syntax so figured I should stay away for now. I'm fine with `string.Concat()` but don't much care for `string.Format()` if it can be avoided - find it's too easy to transpose the format indexes and arguments later on.